src/tx-format/transaction.md

```diff
@@ -104,14 +104,14 @@ The receipts root `receiptsRoot` is the root of the [binary Merkle tree](../prot
 |------------------------|-----------------------------|---------------------------------------------------|
 | `bytecodeLength`       | `uint64`                    | Contract bytecode length, in instructions.        |
 | `bytecodeWitnessIndex` | `uint16`                    | Witness index of contract bytecode to create.     |
-| `policyTypes`          | `uint32`                    | Bitfield of used policy types.                    |
+| `salt`                 | `byte[32]`                  | Salt.                                             |
 | `storageSlotsCount`    | `uint64`                    | Number of storage slots to initialize.            |
+| `policyTypes`          | `uint32`                    | Bitfield of used policy types.                    |
 | `inputsCount`          | `uint16`                    | Number of inputs.                                 |
 | `outputsCount`         | `uint16`                    | Number of outputs.                                |
 | `witnessesCount`       | `uint16`                    | Number of witnesses.                              |
-| `salt`                 | `byte[32]`                  | Salt.                                             |
-| `policies`             | [Policy](./policy.md)`[]`   | List of policies.                                 |
 | `storageSlots`         | `(byte[32], byte[32]])[]`   | List of storage slots to initialize (key, value). |
```

Comment:
I'm kinda confused by how we are specifying the fields of `TransactionCreate` It looks like the first 8 here match what we have in `fuel-vm`, but with `witnessCount` instead of `witness`. But here we also have 5 more that are duplicates.

I guess I'm trying to understand why we have `inputsCount` AND `inputs`, etc, in the specs, but just `inputs` in the code.

Reply:
In the canonical serialization and deserialization, we have a static parts and a dynamic parts. The static part is always the same regardless of the content. In the case of transactions, you always know that unique fields are static, and that is why they are at the top. The dynamic data always mostly lives at the end.

That schema allows us to get the main information about the transaction without deserializing the whole transaction.

An example of the `canonical::Serialize` trait definition:

<img width="1023" alt="image" src="https://github.com/FuelLabs/fuel-specs/assets/18346821/aec10e70-b534-470f-8bbe-c5bf99e8a9ef">